ic-cdk/src/api.rs

```diff
@@ -356,6 +356,22 @@ pub fn stable_read(offset: u64, buf: &mut [u8]) {
     }
 }
 
+/// Gets the public key (a DER-encoded BLS key) of the root key of this instance of the Internet Computer Protocol.
+///
+/// # Note
+///
+/// This traps in non-replicated mode.
+pub fn root_key() -> Principal {
```

Comment:
Why is this function returning a principal? The root key is a Vec<u8>, not a principal.

Reply:
Fixed